mcap_vendor/CMakeLists.txt

```
@@ -35,48 +34,37 @@ macro(build_mcap_vendor)
   file(GLOB _lz4_srcs
     ${lz4_SOURCE_DIR}/lib/*.c)
 
-  add_library(
-    mcap SHARED
+  add_library(${PROJECT_NAME} SHARED
     src/main.cpp
     ${_lz4_srcs}
   )
 
   set(_mcap_include_dir ${mcap_SOURCE_DIR}/cpp/mcap/include)
 
-  target_include_directories(mcap SYSTEM PRIVATE
+  target_include_directories(${PROJECT_NAME} SYSTEM PRIVATE
     ${lz4_SOURCE_DIR}/lib
   )
-  target_include_directories(mcap SYSTEM PUBLIC
+  target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
     "$<BUILD_INTERFACE:${_mcap_include_dir}>"
     "$<INSTALL_INTERFACE:include>"
   )
-  ament_target_dependencies(mcap zstd)
+  ament_target_dependencies(${PROJECT_NAME} zstd)
 
   install(
-    DIRECTORY
-      ${_mcap_include_dir}/mcap
-    DESTINATION
-      ${CMAKE_INSTALL_PREFIX}/include
+    DIRECTORY ${_mcap_include_dir}/mcap
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include
   )
 
-  install(TARGETS mcap EXPORT export_mcap)
+  install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME})
 endmacro()
 
 ## Call vendor macro
 build_mcap_vendor()
 
-ament_export_targets(export_mcap HAS_LIBRARY_TARGET)
-
+ament_export_include_directories(${_mcap_include_dir})
```

And we should correspondingly change this to:

```suggestion
ament_export_include_directories(include/${PROJECT_NAME})
```

---

I agree that we need to change it to the include from install folder rather than source  of the mcap.
Although will need to decide where to install `mcap` header. I will make changes after resolving discussion in previous comment https://github.com/ros-tooling/rosbag2_storage_mcap/pull/62/files#r1002289155

---

Fixed.